Add minimal support for Renesas RZ/A2M EVK

binhnguyen2434 opened the pull request:
Add minimal support for RZ/A2M EVK as following:

| Interface   | Controller | Driver/Component |
| ------------- | ------------- | ------------- |
| GPIO | on-chip | gpio |
| OSTM | on-chip | timer |
| Pinctrl | on-chip | pinctrl |
| UART | on-chip | serial |

Document: https://builds.zephyrproject.io/zephyr/pr/88888/docs/boards/renesas/rza2m_evk/doc/index.html

This SoC does not utilize the FSP driver in [ hal_renesas ](https://github.com/zephyrproject-rtos/hal_renesas/) because it is not supported; instead, the driver is developed directly within the Zephyr tree.

Commit:
soc: renesas: Add support for Renesas RZ/A2M

Add support for Renesas RZ/A2M

Signed-off-by: Hoang Nguyen <hoang.nguyen.jx@bp.renesas.com>
Signed-off-by: Binh Nguyen <binh.nguyen.xw@renesas.com>

Commit:
drivers: clock control: Initial support for RZ/A2M

Add clock control support for RZ/A2M

Signed-off-by: Hoang Nguyen <hoang.nguyen.jx@bp.renesas.com>
Signed-off-by: Binh Nguyen <binh.nguyen.xw@renesas.com>

Commit:
drivers: timer: Add initial support for RZ/A2M

Add timer support for RZ/A2M

Signed-off-by: Hoang Nguyen <hoang.nguyen.jx@bp.renesas.com>
Signed-off-by: Binh Nguyen <binh.nguyen.xw@renesas.com>

Commit:
drivers: pinctrl: Add initial support for RZ/A2M

Add pinctrl support for RZ/A2M

Signed-off-by: Hieu Nguyen <hieu.nguyen.ym@bp.renesas.com>
Signed-off-by: Binh Nguyen <binh.nguyen.xw@renesas.com>
Signed-off-by: Hoang Nguyen <hoang.nguyen.jx@bp.renesas.com>

Commit:
drivers: gpio: Add support for RZ/A2M

Add gpio support for RZ/A2M

Signed-off-by: Hoang Nguyen <hoang.nguyen.jx@bp.renesas.com>
Signed-off-by: Binh Nguyen <binh.nguyen.xw@renesas.com>

Commit:
boards: renesas: Add minimal support for Renesas RZ/A2M EVK

Add minimal support for Renesas RZ/A2M EVK

Signed-off-by: Hoang Nguyen <hoang.nguyen.jx@bp.renesas.com>
Signed-off-by: Binh Nguyen <binh.nguyen.xw@renesas.com>

nordicjm commented on soc/renesas/rz/rza2m/Kconfig.defconfig:
```diff
@@ -0,0 +1,28 @@
+
+if SOC_SERIES_RZA2M
+
+config SYS_CLOCK_HW_CYCLES_PER_SEC
+	default 66000000
```
needs to come from dts property

binhnguyen2434 replied:
@nordicjm Thanks for reviewing.
Updated to get clock freq from dts.

nordicjm commented on dts/arm/renesas/rz/rza/r7s9210.dtsi:
```diff
@@ -0,0 +1,56 @@
+	osc: osc {
+		#clock-cells = <0>;
+		compatible = "fixed-clock";
+		/* Value must be set by board */
+		clock-frequency = <0>;
```
so why is this here? If it needs to be set, omit it so it throws an error if a board does not have it set

binhnguyen2434 replied:
Removed redundant clock freq.

nordicjm replied:
still here, remove from original commit

binhnguyen2434 replied:
@nordicjm 
Thanks, I removed it from original commit.

nordicjm commented on boards/renesas/rza2m_evk/rza2m_evk.yaml:
```diff
@@ -0,0 +1,10 @@
+  - zephyr
+  - gnuarmemb
+supported:
+
```
??

nordicjm commented on boards/renesas/rza2m_evk/rza2m_evk.yaml:
```diff
@@ -6,5 +6,5 @@ toolchain:
   - gnuarmemb
 supported:
-
+  - uart
```
so omit this empty part in the original commit and add the whole part in the commit adding UART

binhnguyen2434 replied:
I moved this change to board's commit.

Commit:
samples: subsys: llext: shell_loader: Exclude Renesas RZ/A2M

RZ/A2M requires MMU enabled. This sample disables MMU, so exclude the
platform.

Signed-off-by: Hoang Nguyen <hoang.nguyen.jx@bp.renesas.com>
Signed-off-by: Binh Nguyen <binh.nguyen.xw@renesas.com>

Commit:
drivers: serial: Add support for RZ/A2M

Add serial support for RZ/A2M

Signed-off-by: Hoang Nguyen <hoang.nguyen.jx@bp.renesas.com>
Signed-off-by: Binh Nguyen <binh.nguyen.xw@renesas.com>
